### PR TITLE
Fix vamc station col

### PR DESCRIPTION
**Pull request by gigxz**
VAMC station column is an int, but it needs to be a string. Some station numbers have non numeric chars.

### CODE REVIEW

**Comment by gigxz** [comment]
@eanders @awisema do we have any migration I can look at that does the above steps? figured this warning has come up so we must have done this before.

**Comment by eanders** [comment]
You can do:
1. Create new column, copy data from the old to the new, then rename the old column and the new column so that the new column has the old column name.
OR
2. Use `StrongMigrations.disable_check(:check_name)` where check_name can be found [here](https://github.com/ankane/strong_migrations/blob/master/lib/strong_migrations/error_messages.rb).  At the end of the migration just add an ensure with `enable_check(:check_name)`

**Comment by eanders** [comment]
[Here's an example.](https://github.com/greenriver/hmis-warehouse/blob/6a61a52506f9f1116161e7116aa9fe4788506652/db/migrate/20230807142127_create_hmis_user_groups.rb)

**Comment by gigxz** [comment]
I keep trying different things but the migration just hangs.
The `update_all` I have in here works just fine in the console, but it looks like thats where its freezing up.  Do you have any other suggestions @eanders or @awisema ? you mentioned arel, and I tried this with qualified_column, and it didnt like that. Were you thinking of `Arel::UpdateManager` or some other approach? I'm not familiar

It just freezes after this:
```
   (1.4ms)  ALTER TABLE "Enrollment" ADD "VAMCStation_new" character varying /*application:BostonHmis,line:/db/warehouse/migrate/20230808144340_change_vamc_station_to_string.rb:6:in `change'*/
  ↳ /app/db/warehouse/migrate/20230808144340_change_vamc_station_to_string.rb:6:in `change'
Query Trace:
      db/warehouse/migrate/20230808144340_change_vamc_station_to_string.rb:6:in `change'
      /bundle/ruby/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
      /bundle/ruby/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
      /bundle/ruby/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
      /bundle/ruby/3.1.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
  ↳ /app/db/warehouse/migrate/20230808144340_change_vamc_station_to_string.rb:6:in `change'
   -> 0.0028s
```

**Comment by gigxz** [comment]
This still needs to be fixed

**Comment by awisema** [comment]
I think I know what is going on here -- migrations are run inside transactions so you are deadlocking on line 7.

Two ideas: either try breaking this into 3 migrations: add the column, do the copy, clean up the columns... Or, you can put a `USING` on `change_column`... I think it would look something like:

`change_column :Enrollment, :VAMCStation, "varchar USING CAST ( 'VAMCStation' AS varchar )"`

**Comment by gigxz** [comment]
Ahh that makes sense. Thank you, I will try that

**Comment by gigxz** [comment]
Okay, I think this is ok now. I removed the Bi::ViewMaintainer stuff since I think it was a red herring..

**Review by awisema — APPROVED** [review]

**Comment by gigxz** [comment]
@eanders can i merge this?

File: db/warehouse/migrate/20230828180743_copy_over_vamc_column.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class CopyOverVamcColumn < ActiveRecord::Migration[6.1]
+  def change
```
Comment (eanders) [inline db/warehouse/migrate/20230828180743_copy_over_vamc_column.rb]: This is probably fine, but it should probably be:
```suggestion
  def up
```

**Review by eanders — APPROVED** [review]
Just the one note that the migration is not really reversible.  Otherwise this looks good!

**Pull request merged**